src/inc/regdisp.h

```diff
@@ -396,7 +396,7 @@ inline void FillRegDisplay(const PREGDISPLAY pRD, PT_CONTEXT pctx, PT_CONTEXT pC
 #elif defined(_TARGET_X86_) // _TARGET_ARM_
     for (int i = 0; i < 7; i++)
     {
-        *(&pRD->ctxPtrsOne.Esi + i) = (&pctx->Esi + i);
+        *(&pRD->ctxPtrsOne.Esi + i) = (&pRD->pCurrentContext->Esi + i);
```

Comment:
As for why there is the pContext in the REGDISPLAY, from what I can see from the code, it is used for debugger purposes. It keeps pointer to the context from which the REGDISPLAY was initilized and it does that only until we unwind from that context in one of the XXX::UpdateRegDisplayHelper functions and then it is nulled.